specifyweb/frontend/js_src/lib/components/QueryBuilder/Results.tsx

```diff
@@ -202,7 +202,7 @@ export function QueryResults(props: Props): JSX.Element {
         Array.isArray(results) &&
         Array.isArray(loadedResults) &&
         results.length > 0 &&
-        typeof fetchResults === 'function' ? (
+        typeof fetchResults === 'function' && totalCount !== 0 ? (
```

Comment:
how is `totalCount !== 0` different from the `results.length > 0` condition that was already here (one line above)?

only one of those should be needed, unless something fishy is going on here - can you investigate? @CarolineDenis

Reply:
because sometimes results = [undefined] instead of []

Reply:
I'm looking into it now... 
handleSetResults is being with results=undefined and it shouldn't be called at all

Reply:
seems to work every time

Reply:
good catch - thanks for resolving this issue the proper way rather than adding a hack